### ISSUE TITLE
Still working on it?

### ISSUE THREAD

**Issue by jonas-w** [Animeshz/EgisTec-EH575#2]
Hey, by any chance is there a way to install this repo. Or are you still working on it? I've found your comment in the libfprint forum. But since then nothing changed on this repo. 

Thank you in advance

**Comment by Animeshz** [comment]
Actually I found the sequences and put some debugging details here https://github.com/Animeshz/EgisTec-EH575/discussions/1#discussioncomment-1142115 lately on August, but I had exams coming soon, so I just started to put it on the libprint since I saw the 0570 just merged back then, so I hope matching algorithm was working somewhat fine for that small sensor (0570: 114x57, this one is 103x52)...

Started the WIP PR [libfprint#317](https://gitlab.freedesktop.org/libfprint/libfprint/-/merge_requests/317) (ignore those images archive it was problem in implementation, it takes images perfectly now, you can test), but the next day I got notified about the exam timings, so I was partially online or basically offline till 3rd of October (2nd last day from now, I had 4 exam in series). I had problems with libfprint swipe sensor implementation due to a lot of false rejection rates. Mostly working capturing c-code with those sequences and few-retry logic is here [gitlab:Animeshz/libfprint](https://gitlab.freedesktop.org/Animeshz/libfprint). Just updated it to the latest upstream.

I still have not much time (I have my linux setup messed up rn) for manually testing different values for ***swipe sensor*** implementation (as libfprint can't handle small images for touch sensing, same for 0570). The diff from original repo can be seen [on web here](https://gitlab.freedesktop.org/Animeshz/libfprint/-/compare/master...master?from_project_id=124) to easily see the changes made. If anyone can work further for finding optimum values for the height/bz3_threshold or anything else that is making these false rejections and PR it maybe from fork it'll be appreciated :-)

[Partially documented on libfprint repository](https://gitlab.freedesktop.org/libfprint/libfprint/-/blob/master/INSTALL), these commands will easily let you build and test the current state and your changes if you do.
```bash
# build
# packages might need (names on void-linux might differ on other): meson cmake libgusb libgusb-devel gobject-introspection nss-devel glib-devel gtk-doc gdb valgrind
meson --buildtype=release . _build    # first time
ninja -C _build                       # every changes made

cd _build/examples

# test image captured
sudo ./img-capture  # capture into finger.pgm file

# test enroll and verify
sudo ./enroll
sudo ./verify
```

**Comment by jonas-w** [comment]
At first, thank you for your fast reply, hope someone is going to work on this project too.
Because i'm such an idiot. I tried it multiple times until i realized, it doesn't find the driver because i have an EH57E and not an EH575... are there similarities between them? will this driver maybe work on the EH57E or is this a whole another story?

**Comment by Animeshz** [comment]
@jonas-w umm,

1. if you have windows, you can use usbpcap (enable testsigning as documented in docs) to collect the logs when using the scanner through original driver.
2. If you have a windows in vm (vmware can passthru usb), add vmware usb log option (see below) so it collect the logs, connect the fingerprint device, update the driver if haven't using device manager, use the fingerprint actions like setting / using it, you can directly see it in vmware.log file for the sequences, but its cluttered, i fixed the old broken script that I found of web archive and put it on #1 thread you can use to extract vm logs to standard pcap wireshark file as well if you like, it'll properly distinguish between meta-data and the data-layer in the transfer packets.
```
# vmname.vmx
monitor = "debug"
usb.analyzer.enable = TRUE
usb.analyzer.maxLine = 8192
```

The related sequences are available at [egis0575.h](https://gitlab.freedesktop.org/Animeshz/libfprint/-/blob/master/libfprint/drivers/egis0575.h) file in the above diffs/repository, also on the archive present in #1. If both yours and heres matches, you can add a new vid/pid at [this position](https://gitlab.freedesktop.org/Animeshz/libfprint/-/blob/master/libfprint/drivers/egis0575.c#L391) and it should work fine.

**Note:** you can just add it to that line and see if it works, but I'd just recommend to obtain the original logs from transfer packets using original driver :)) Also see if it matches with eh0570 that is already merged.

**Comment by Animeshz** [comment]
@jonas-w if it says `Image device reported finger status: on`, then atleast one finger present check is success (Standard deviation of pixels >= 18) SD usually roam around 20-40 sometimes upto 50 for very sharp image.

So the sequences work to capture the image, it would work I guess, maybe you tap the sensor? Its implemented as a swipe sensor similar to 0570 because libfprint can't deal with small images, try swiping the finger slowly while putting it there in the sensor.

**Comment by TmDobbin** [comment]
Hello, apologies for posting to a fairly old issue, but I have the EH575, and I'd love to see it working under linux. I tried building it, but meson throws some errors at me (I'm using Linux Mint 20.3) - and from what I can see in this issue, it wouldn't even work if I could get it to build properly? I would also really love to help out here, but the majority of my experience is in python, which I don't think would be super helpful, and I also have no experience with fingerprint sensors, lol. Would appreciate an update on this! :)

**Comment by Animeshz** [comment]
@TmDobbin it'll work only if libfprint can handle such small images (103x52), which it doesn't ([libfrpint#271](https://gitlab.freedesktop.org/libfprint/libfprint/-/issues/271) [libfprint#272](https://gitlab.freedesktop.org/libfprint/libfprint/-/issues/271)) . idk if any foss algorithm exist which deal with it :(

**Comment by TmDobbin** [comment]
First of all, apologies for the late response! I've been very busy lately. 

Second of all, that's a big shame. Is there some way that the images taken by the reader can be dumped to a directory under linux? Just out of interest really. Is swiping across the sensor and merging the images into a larger one so libfprint can handle it a viable option at all?

**Comment by TmDobbin** [comment]
Quick update by the way, I finally managed to get it to compile, using sudo ./img-capture does nothing unless my thumb (only my thumb, fingers don't seem to work) is on the sensor, in which case it reads `(process:27946): libfprint-SSM-DEBUG: 17:00:16.762: [egis0575] SM_STATES_NUM entering state 2` over and over.

**Comment by Animeshz** [comment]
@TmDobbin sorry for late reply too

> Is there some way that the images taken by the reader can be dumped to a directory under linux?

Yeah, as you found, [libfprint/examples/img_capture.c](https://github.com/iafilatov/libfprint/blob/master/examples/img_capture.c) demonstrate exactly this, `fp_img_save_to_file` can be used to save it as a file, you may want to apply the patch before build that does this. It'll be easy if your package manager [brings a template and has some automated patching ability](https://github.com/void-linux/void-packages/tree/master/srcpkgs/libfprint).

> using sudo ./img-capture does nothing unless my thumb (only my thumb, fingers don't seem to work) is on the sensor

In [libfprint#317](https://gitlab.freedesktop.org/libfprint/libfprint/-/merge_requests/317) the sensor is implemented as a swipe one actually, if you don't swipe, it won't be probably able to merge the images. Yet still some merges are quite unreliable due to small size, and has high false rejection for same reason.

**Comment by TmDobbin** [comment]
No problem! 
Thanks for the info, it looks like any kind of development on this is out of my reach, so I suppose I can only hope someone makes it work at some point lol.

**Comment by topni1** [comment]
Hi,
I am following this issue for a while now. I tried out the driver here https://gitlab.freedesktop.org/Animeshz/libfprint a while ago, but it didnt work for me (error "all zeros recieved") so i tried to understand the code and also looked at the packages my sensor produced with the windows driver. After some time I got a little python script that worked with my sensor (I think the initial packages were wrong).

I rewrote your driver and I now have a version that produces ok looking fingerprints and I can enroll and sometimes it even verify my fingerprint. I dont know if my version is an improvement to yours, because yours didnt work for me.

Im not really familiar with git and i dont know how and where i should publish this. So if someone is willing to try my version to see if it works on other PCs too and look if it maybe works better than the current version I would happily take the help :)

**Comment by SkiFire13** [comment]
@topni1 I'm kinda in your same situation, the old one didn't work, I changed the packages and now it kinda works, though I don't think my images are that good and in fact I couldn't get a verify to work.

**Comment by Animeshz** [comment]
@topni1 nice, if you're able to get fingerprints matched then it'd surely be an improvement, because in my case it was matching only when swiped exactly like when enrolled, and with not much force.

Not sure what packages we are talking about and why did zeroes came, because the sequence was working at my side for `1c7a:0575`, did you had `17ca:0575E`? Anyway good job for fixing any unexpected behavior, if I've missed anything.

You can put your code on [your fork at freedesktop's gitlab](https://gitlab.freedesktop.org/libfprint/libfprint/-/forks), and let others try too. If its very small you can place them as [gist](https://gist.github.com) temporarily (by copy pasting over web) too if you feel you're not confident in git. I've actually moved over to framework laptop, and given acer to my uncle, which currently runs windows, so I won't be able to provide much help in testing yours, people here may be willing to test them :)

**Comment by topni1** [comment]
Ok I have copied everything into a public github repo [here](https://github.com/topni1/LibfprintEgis0575driver). I will do it as a gitlab branch/fork later.

lsusb gives me "ID 1c7a:0575 LighTuning Technology Inc. EgisTec EH575" this looks like yours.

I wouldnt get to excited, it does only verify sometimes, if the enrollment was perfect and the verification too.

**Comment by topni1** [comment]
Ok quick update: 
I put everything in a gitlab fork of libfprint (took so long because i got errors from the tests).
[https://gitlab.freedesktop.org/topni1/libfprint](https://gitlab.freedesktop.org/topni1/libfprint)
I also tweaked some variables and it verifies now reliably with a long, steady swipe of the whole fingertip.
If someone tries this out please write me somewhere if it worked. I will probably make the merge request in the following weeks, would really like to know if it works for someone else too before doing that :)

**Comment by ajboni** [comment]
> Ok quick update: I put everything in a gitlab fork of libfprint (took so long because i got errors from the tests). https://gitlab.freedesktop.org/topni1/libfprint I also tweaked some variables and it verifies now reliably with a long, steady swipe of the whole fingertip. If someone tries this out please write me somewhere if it worked. I will probably make the merge request in the following weeks, would really like to know if it works for someone else too before doing that :)

Hey I have a machine to test this, but I'm a little out of the loop, can you describe what steps to follow?

Like this?
```
# build
# packages might need (names on void-linux might differ on other): meson cmake libgusb libgusb-devel gobject-introspection nss-devel glib-devel gtk-doc gdb valgrind
meson --buildtype=release . _build    # first time
ninja -C _build                       # every changes made

cd _build/examples

# test image captured
sudo ./img-capture  # capture into finger.pgm file

# test enroll and verify
sudo ./enroll
sudo ./verify
```

**Comment by jonas-w** [comment]
Yes exactly that's how i compiled it, but i did have the wrong Fingerprint scanner. But mine doesn't seem to differ that much, so it partially worked but it isn't meant to work for mine, so that's the way you could give it a try!

**Comment by ajboni** [comment]
This is my device
`Bus 003 Device 002: ID 1c7a:0575 LighTuning Technology Inc. EgisTec EH575`

The script advances automatically as if its detecting input, without me actually scanning the finger...
This is the log for ./enroll, hope it helps!

```
sudo ./enroll
This program will enroll the selected finger overwriting any print for the same finger that was enrolled previously. Fingerprint updates without erasing old data are possible on devices supporting that. Ctrl+C interrupts program execution.
Choose the finger to enroll:
  [0] left thumb
  [1] left index
  [2] left middle
  [3] left ring
  [4] left little
  [5] right thumb
  [6] right index
  [7] right middle
  [8] right ring
  [9] right little
> 0
(process:7507): libfprint-context-DEBUG: 15:47:27.008: No driver found for USB device 1D6B:0003
(process:7507): libfprint-context-DEBUG: 15:47:27.008: No driver found for USB device 1D6B:0002
(process:7507): libfprint-context-DEBUG: 15:47:27.008: No driver found for USB device 1D6B:0003
(process:7507): libfprint-context-DEBUG: 15:47:27.008: No driver found for USB device 8087:0029
(process:7507): libfprint-context-DEBUG: 15:47:27.008: No driver found for USB device 0408:A094
(process:7507): libfprint-context-DEBUG: 15:47:27.008: No driver found for USB device 1EA7:0064
(process:7507): libfprint-context-DEBUG: 15:47:27.008: No driver found for USB device 1D6B:0002
(process:7507): libfprint-device-DEBUG: 15:47:27.009: Device reported probe completion
(process:7507): libfprint-device-DEBUG: 15:47:27.009: Completing action FPI_DEVICE_ACTION_PROBE in idle!
(process:7507): libfprint-device-DEBUG: 15:47:27.009: Updated temperature model after 0.00 seconds, ratio 0.27 -> 0.27, active 0 -> 0, FP_TEMPERATURE_COLD -> FP_TEMPERATURE_COLD
Selected device 0 (LighTuning Technology Inc. EgisTec EH575) claimed by egis0575 driver
(process:7507): libfprint-egis0575-DEBUG: 15:47:27.125: EGIS INIT
(process:7507): libfprint-SSM-DEBUG: 15:47:27.125: [egis0575] SSM_INIT_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:27.125: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:27.143: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-SSM-DEBUG: 15:47:27.153: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:27.164: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-SSM-DEBUG: 15:47:27.170: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:27.193: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-SSM-DEBUG: 15:47:27.193: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:27.201: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-SSM-DEBUG: 15:47:27.201: [egis0575] SSM_INIT_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:27.201: EGIS INIT DONE
(process:7507): libfprint-image_device-DEBUG: 15:47:27.201: Image device open completed
(process:7507): libfprint-device-DEBUG: 15:47:27.201: Device reported open completion
(process:7507): libfprint-device-DEBUG: 15:47:27.201: Completing action FPI_DEVICE_ACTION_OPEN in idle!
(process:7507): libfprint-device-DEBUG: 15:47:27.201: Updated temperature model after 0.19 seconds, ratio 0.27 -> 0.27, active 0 -> 0, FP_TEMPERATURE_COLD -> FP_TEMPERATURE_COLD
Opened device.
The device supports fingerprint updates.
Should an existing fingerprint be updated instead of being replaced (if present)? Enter Y/y or N/n to make a choice.
y
It's now time to enroll your finger.

You will need to successfully scan your left thumb finger 5 times to complete the process.

Scan your finger now.
(process:7507): libfprint-device-DEBUG: 15:47:29.338: Updated temperature model after 2.14 seconds, ratio 0.27 -> 0.27, active 1 -> 1, FP_TEMPERATURE_COLD -> FP_TEMPERATURE_COLD
(process:7507): libfprint-image_device-DEBUG: 15:47:29.338: Activating image device
(process:7507): libfprint-image_device-DEBUG: 15:47:29.338: Image device internal state change from FPI_IMAGE_DEVICE_STATE_INACTIVE to FPI_IMAGE_DEVICE_STATE_ACTIVATING
(process:7507): libfprint-egis0575-DEBUG: 15:47:29.338: EGIS START
(process:7507): libfprint-SSM-DEBUG: 15:47:29.338: [egis0575] IMG_SSM_DONE entering state 0
(process:7507): libfprint-egis0575-DEBUG: 15:47:29.338: EGIS START DONE
(process:7507): libfprint-image_device-DEBUG: 15:47:29.338: Image device activation completed
(process:7507): libfprint-image_device-DEBUG: 15:47:29.338: Image device internal state change from FPI_IMAGE_DEVICE_STATE_ACTIVATING to FPI_IMAGE_DEVICE_STATE_IDLE
(process:7507): libfprint-image_device-DEBUG: 15:47:29.338: Image device internal state change from FPI_IMAGE_DEVICE_STATE_IDLE to FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_ON
(process:7507): libfprint-device-DEBUG: 15:47:29.338: Device reported finger status change: FP_FINGER_STATUS_NEEDED
(process:7507): libfprint-device-DEBUG: 15:47:29.338: Updated temperature model after 0.00 seconds, ratio 0.27 -> 0.27, active 1 -> 1, FP_TEMPERATURE_COLD -> FP_TEMPERATURE_COLD
(process:7507): libfprint-SSM-DEBUG: 15:47:29.339: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:29.359: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:29.364: 579.233196
(process:7507): libfprint-SSM-DEBUG: 15:47:29.365: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:29.370: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:29.376: 578.712285
(process:7507): libfprint-SSM-DEBUG: 15:47:29.426: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:29.432: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:29.437: 577.874907
(process:7507): libfprint-SSM-DEBUG: 15:47:29.488: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:29.495: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:29.500: 576.112584
(process:7507): libfprint-SSM-DEBUG: 15:47:29.551: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:29.558: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:29.563: 575.460045
(process:7507): libfprint-SSM-DEBUG: 15:47:29.614: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:29.622: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:29.627: 576.469567
(process:7507): libfprint-SSM-DEBUG: 15:47:29.678: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:29.685: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:29.690: 576.663742
(process:7507): libfprint-device-DEBUG: 15:47:29.723: Updated temperature model after 0.39 seconds, ratio 0.27 -> 0.27, active 1 -> 1, FP_TEMPERATURE_COLD -> FP_TEMPERATURE_WARM
(process:7507): libfprint-SSM-DEBUG: 15:47:29.741: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:29.750: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:29.755: 576.796490
(process:7507): libfprint-SSM-DEBUG: 15:47:29.805: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:29.812: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:29.817: 576.884802
(process:7507): libfprint-SSM-DEBUG: 15:47:29.867: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:29.874: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:29.880: 577.648805
(process:7507): libfprint-SSM-DEBUG: 15:47:29.880: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:29.887: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-device-DEBUG: 15:47:29.887: Device reported finger status change: FP_FINGER_STATUS_NEEDED | FP_FINGER_STATUS_PRESENT
(process:7507): libfprint-image_device-DEBUG: 15:47:29.887: Image device reported finger status: on
(process:7507): libfprint-image_device-DEBUG: 15:47:29.887: Image device internal state change from FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_ON to FPI_IMAGE_DEVICE_STATE_CAPTURE
(process:7507): libfprint-assembling-DEBUG: 15:47:30.083: calc delta completed in 0.196718 secs
(process:7507): libfprint-assembling-DEBUG: 15:47:30.245: calc delta completed in 0.161755 secs
(process:7507): libfprint-assembling-DEBUG: 15:47:30.245: errors: 132019 rev: 131606
(process:7507): libfprint-assembling-DEBUG: 15:47:30.245: height is -387
(process:7507): libfprint-image_device-DEBUG: 15:47:30.246: Image device captured an image
(process:7507): libfprint-image_device-DEBUG: 15:47:30.246: Image device internal state change from FPI_IMAGE_DEVICE_STATE_CAPTURE to FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_OFF
(process:7507): libfprint-device-DEBUG: 15:47:30.246: Device reported finger status change: FP_FINGER_STATUS_PRESENT
(process:7507): libfprint-device-DEBUG: 15:47:30.246: Device reported finger status change: FP_FINGER_STATUS_NONE
(process:7507): libfprint-image_device-DEBUG: 15:47:30.246: Image device reported finger status: off
(process:7507): libfprint-image_device-DEBUG: 15:47:30.246: Image device internal state change from FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_OFF to FPI_IMAGE_DEVICE_STATE_IDLE
(process:7507): libfprint-SSM-DEBUG: 15:47:30.246: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:30.265: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:30.270: 576.830284
(process:7507): libfprint-SSM-DEBUG: 15:47:30.270: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:30.276: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:30.281: 575.482450
(process:7507): libfprint-image-DEBUG: 15:47:30.297: Minutiae scan completed in 0.051272 secs
(process:7507): libfprint-device-DEBUG: 15:47:30.297: Device reported enroll progress, reported 1 of 5 have been completed
(process:7507): libfprint-DEBUG: 15:47:30.298: written to 'enrolled.pgm'
Wrote scanned image to enrolled.pgm
Enroll stage 1 of 5 passed. Yay!
(process:7507): libfprint-image_device-DEBUG: 15:47:30.298: Image device internal state change from FPI_IMAGE_DEVICE_STATE_IDLE to FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_ON
(process:7507): libfprint-device-DEBUG: 15:47:30.298: Device reported finger status change: FP_FINGER_STATUS_NEEDED
(process:7507): libfprint-SSM-DEBUG: 15:47:30.332: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:30.338: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:30.344: 575.407767
(process:7507): libfprint-SSM-DEBUG: 15:47:30.394: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:30.401: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:30.406: 579.126587
(process:7507): libfprint-SSM-DEBUG: 15:47:30.457: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:30.465: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:30.470: 579.767550
(process:7507): libfprint-SSM-DEBUG: 15:47:30.520: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:30.527: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:30.532: 575.736184
(process:7507): libfprint-SSM-DEBUG: 15:47:30.583: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:30.589: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:30.595: 577.665795
(process:7507): libfprint-SSM-DEBUG: 15:47:30.645: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:30.652: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:30.657: 577.948842
(process:7507): libfprint-SSM-DEBUG: 15:47:30.708: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:30.719: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:30.724: 577.444922
(process:7507): libfprint-SSM-DEBUG: 15:47:30.775: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:30.784: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:30.789: 576.171397
(process:7507): libfprint-SSM-DEBUG: 15:47:30.789: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:30.796: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-device-DEBUG: 15:47:30.796: Device reported finger status change: FP_FINGER_STATUS_NEEDED | FP_FINGER_STATUS_PRESENT
(process:7507): libfprint-image_device-DEBUG: 15:47:30.796: Image device reported finger status: on
(process:7507): libfprint-image_device-DEBUG: 15:47:30.796: Image device internal state change from FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_ON to FPI_IMAGE_DEVICE_STATE_CAPTURE
(process:7507): libfprint-assembling-DEBUG: 15:47:30.993: calc delta completed in 0.196795 secs
(process:7507): libfprint-assembling-DEBUG: 15:47:31.154: calc delta completed in 0.161401 secs
(process:7507): libfprint-assembling-DEBUG: 15:47:31.154: errors: 131070 rev: 130420
(process:7507): libfprint-assembling-DEBUG: 15:47:31.154: height is -387
(process:7507): libfprint-image_device-DEBUG: 15:47:31.154: Image device captured an image
(process:7507): libfprint-image_device-DEBUG: 15:47:31.154: Image device internal state change from FPI_IMAGE_DEVICE_STATE_CAPTURE to FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_OFF
(process:7507): libfprint-device-DEBUG: 15:47:31.154: Device reported finger status change: FP_FINGER_STATUS_PRESENT
(process:7507): libfprint-device-DEBUG: 15:47:31.154: Device reported finger status change: FP_FINGER_STATUS_NONE
(process:7507): libfprint-image_device-DEBUG: 15:47:31.154: Image device reported finger status: off
(process:7507): libfprint-image_device-DEBUG: 15:47:31.154: Image device internal state change from FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_OFF to FPI_IMAGE_DEVICE_STATE_IDLE
(process:7507): libfprint-SSM-DEBUG: 15:47:31.155: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:31.173: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:31.179: 578.958178
(process:7507): libfprint-SSM-DEBUG: 15:47:31.179: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:31.185: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:31.190: 578.711352
(process:7507): libfprint-image-DEBUG: 15:47:31.203: Minutiae scan completed in 0.048772 secs
(process:7507): libfprint-device-DEBUG: 15:47:31.204: Device reported enroll progress, reported 2 of 5 have been completed
(process:7507): libfprint-DEBUG: 15:47:31.204: written to 'enrolled.pgm'
Wrote scanned image to enrolled.pgm
Enroll stage 2 of 5 passed. Yay!
(process:7507): libfprint-image_device-DEBUG: 15:47:31.204: Image device internal state change from FPI_IMAGE_DEVICE_STATE_IDLE to FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_ON
(process:7507): libfprint-device-DEBUG: 15:47:31.204: Device reported finger status change: FP_FINGER_STATUS_NEEDED
(process:7507): libfprint-SSM-DEBUG: 15:47:31.241: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:31.248: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:31.253: 579.959298
(process:7507): libfprint-SSM-DEBUG: 15:47:31.304: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:31.311: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:31.317: 577.458551
(process:7507): libfprint-SSM-DEBUG: 15:47:31.367: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:31.374: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:31.380: 577.020911
(process:7507): libfprint-SSM-DEBUG: 15:47:31.430: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:31.437: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:31.442: 578.072816
(process:7507): libfprint-SSM-DEBUG: 15:47:31.493: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:31.500: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:31.505: 577.523898
(process:7507): libfprint-SSM-DEBUG: 15:47:31.555: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:31.562: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:31.568: 581.328043
(process:7507): libfprint-SSM-DEBUG: 15:47:31.618: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:31.626: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:31.631: 578.886669
(process:7507): libfprint-SSM-DEBUG: 15:47:31.681: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:31.691: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:31.696: 578.576736
(process:7507): libfprint-SSM-DEBUG: 15:47:31.696: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:31.703: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-device-DEBUG: 15:47:31.704: Device reported finger status change: FP_FINGER_STATUS_NEEDED | FP_FINGER_STATUS_PRESENT
(process:7507): libfprint-image_device-DEBUG: 15:47:31.704: Image device reported finger status: on
(process:7507): libfprint-image_device-DEBUG: 15:47:31.704: Image device internal state change from FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_ON to FPI_IMAGE_DEVICE_STATE_CAPTURE
(process:7507): libfprint-assembling-DEBUG: 15:47:31.900: calc delta completed in 0.196721 secs
(process:7507): libfprint-assembling-DEBUG: 15:47:32.062: calc delta completed in 0.161739 secs
(process:7507): libfprint-assembling-DEBUG: 15:47:32.062: errors: 130467 rev: 130048
(process:7507): libfprint-assembling-DEBUG: 15:47:32.062: height is -387
(process:7507): libfprint-image_device-DEBUG: 15:47:32.062: Image device captured an image
(process:7507): libfprint-image_device-DEBUG: 15:47:32.062: Image device internal state change from FPI_IMAGE_DEVICE_STATE_CAPTURE to FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_OFF
(process:7507): libfprint-device-DEBUG: 15:47:32.062: Device reported finger status change: FP_FINGER_STATUS_PRESENT
(process:7507): libfprint-device-DEBUG: 15:47:32.062: Device reported finger status change: FP_FINGER_STATUS_NONE
(process:7507): libfprint-image_device-DEBUG: 15:47:32.062: Image device reported finger status: off
(process:7507): libfprint-image_device-DEBUG: 15:47:32.062: Image device internal state change from FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_OFF to FPI_IMAGE_DEVICE_STATE_IDLE
(process:7507): libfprint-SSM-DEBUG: 15:47:32.063: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:32.082: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:32.087: 580.471247
(process:7507): libfprint-SSM-DEBUG: 15:47:32.087: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:32.093: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:32.098: 577.415982
(process:7507): libfprint-image-DEBUG: 15:47:32.114: Minutiae scan completed in 0.051490 secs
(process:7507): libfprint-device-DEBUG: 15:47:32.114: Device reported enroll progress, reported 3 of 5 have been completed
(process:7507): libfprint-DEBUG: 15:47:32.114: written to 'enrolled.pgm'
Wrote scanned image to enrolled.pgm
Enroll stage 3 of 5 passed. Yay!
(process:7507): libfprint-image_device-DEBUG: 15:47:32.114: Image device internal state change from FPI_IMAGE_DEVICE_STATE_IDLE to FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_ON
(process:7507): libfprint-device-DEBUG: 15:47:32.115: Device reported finger status change: FP_FINGER_STATUS_NEEDED
(process:7507): libfprint-SSM-DEBUG: 15:47:32.149: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:32.155: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:32.161: 576.863144
(process:7507): libfprint-SSM-DEBUG: 15:47:32.211: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:32.217: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:32.223: 577.769231
(process:7507): libfprint-SSM-DEBUG: 15:47:32.273: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:32.280: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:32.285: 580.051344
(process:7507): libfprint-SSM-DEBUG: 15:47:32.335: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:32.342: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:32.348: 577.833271
(process:7507): libfprint-SSM-DEBUG: 15:47:32.399: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:32.406: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:32.411: 577.723114
(process:7507): libfprint-SSM-DEBUG: 15:47:32.462: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:32.469: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:32.474: 579.148245
(process:7507): libfprint-SSM-DEBUG: 15:47:32.525: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:32.532: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:32.537: 576.178678
(process:7507): libfprint-SSM-DEBUG: 15:47:32.588: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:32.597: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:32.602: 579.391524
(process:7507): libfprint-SSM-DEBUG: 15:47:32.602: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:32.609: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-device-DEBUG: 15:47:32.609: Device reported finger status change: FP_FINGER_STATUS_NEEDED | FP_FINGER_STATUS_PRESENT
(process:7507): libfprint-image_device-DEBUG: 15:47:32.609: Image device reported finger status: on
(process:7507): libfprint-image_device-DEBUG: 15:47:32.609: Image device internal state change from FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_ON to FPI_IMAGE_DEVICE_STATE_CAPTURE
(process:7507): libfprint-assembling-DEBUG: 15:47:32.804: calc delta completed in 0.195465 secs
(process:7507): libfprint-assembling-DEBUG: 15:47:32.966: calc delta completed in 0.161633 secs
(process:7507): libfprint-assembling-DEBUG: 15:47:32.966: errors: 130669 rev: 129958
(process:7507): libfprint-assembling-DEBUG: 15:47:32.966: height is -387
(process:7507): libfprint-image_device-DEBUG: 15:47:32.966: Image device captured an image
(process:7507): libfprint-image_device-DEBUG: 15:47:32.966: Image device internal state change from FPI_IMAGE_DEVICE_STATE_CAPTURE to FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_OFF
(process:7507): libfprint-device-DEBUG: 15:47:32.966: Device reported finger status change: FP_FINGER_STATUS_PRESENT
(process:7507): libfprint-device-DEBUG: 15:47:32.966: Device reported finger status change: FP_FINGER_STATUS_NONE
(process:7507): libfprint-image_device-DEBUG: 15:47:32.966: Image device reported finger status: off
(process:7507): libfprint-image_device-DEBUG: 15:47:32.966: Image device internal state change from FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_OFF to FPI_IMAGE_DEVICE_STATE_IDLE
(process:7507): libfprint-SSM-DEBUG: 15:47:32.967: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:32.985: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:32.990: 580.327483
(process:7507): libfprint-SSM-DEBUG: 15:47:32.990: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:32.996: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:33.001: 581.833084
(process:7507): libfprint-image-DEBUG: 15:47:33.019: Minutiae scan completed in 0.052529 secs
(process:7507): libfprint-device-DEBUG: 15:47:33.019: Device reported enroll progress, reported 4 of 5 have been completed
(process:7507): libfprint-DEBUG: 15:47:33.020: written to 'enrolled.pgm'
Wrote scanned image to enrolled.pgm
Enroll stage 4 of 5 passed. Yay!
(process:7507): libfprint-image_device-DEBUG: 15:47:33.020: Image device internal state change from FPI_IMAGE_DEVICE_STATE_IDLE to FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_ON
(process:7507): libfprint-device-DEBUG: 15:47:33.020: Device reported finger status change: FP_FINGER_STATUS_NEEDED
(process:7507): libfprint-SSM-DEBUG: 15:47:33.052: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:33.059: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:33.064: 581.257842
(process:7507): libfprint-SSM-DEBUG: 15:47:33.115: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:33.121: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:33.127: 578.986744
(process:7507): libfprint-SSM-DEBUG: 15:47:33.177: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:33.184: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:33.189: 580.215459
(process:7507): libfprint-SSM-DEBUG: 15:47:33.240: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:33.248: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:33.254: 578.940814
(process:7507): libfprint-SSM-DEBUG: 15:47:33.304: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:33.313: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:33.318: 575.196602
(process:7507): libfprint-SSM-DEBUG: 15:47:33.368: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:33.378: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:33.383: 581.967326
(process:7507): libfprint-SSM-DEBUG: 15:47:33.434: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:33.440: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:33.446: 579.205937
(process:7507): libfprint-SSM-DEBUG: 15:47:33.496: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:33.505: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:33.510: 579.470314
(process:7507): libfprint-SSM-DEBUG: 15:47:33.510: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:33.517: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-device-DEBUG: 15:47:33.517: Device reported finger status change: FP_FINGER_STATUS_NEEDED | FP_FINGER_STATUS_PRESENT
(process:7507): libfprint-image_device-DEBUG: 15:47:33.517: Image device reported finger status: on
(process:7507): libfprint-image_device-DEBUG: 15:47:33.517: Image device internal state change from FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_ON to FPI_IMAGE_DEVICE_STATE_CAPTURE
(process:7507): libfprint-assembling-DEBUG: 15:47:33.713: calc delta completed in 0.195889 secs
(process:7507): libfprint-assembling-DEBUG: 15:47:33.875: calc delta completed in 0.161997 secs
(process:7507): libfprint-assembling-DEBUG: 15:47:33.875: errors: 131609 rev: 130913
(process:7507): libfprint-assembling-DEBUG: 15:47:33.875: height is -387
(process:7507): libfprint-image_device-DEBUG: 15:47:33.875: Image device captured an image
(process:7507): libfprint-image_device-DEBUG: 15:47:33.875: Image device internal state change from FPI_IMAGE_DEVICE_STATE_CAPTURE to FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_OFF
(process:7507): libfprint-device-DEBUG: 15:47:33.875: Device reported finger status change: FP_FINGER_STATUS_PRESENT
(process:7507): libfprint-device-DEBUG: 15:47:33.875: Device reported finger status change: FP_FINGER_STATUS_NONE
(process:7507): libfprint-image_device-DEBUG: 15:47:33.875: Image device reported finger status: off
(process:7507): libfprint-image_device-DEBUG: 15:47:33.875: Image device internal state change from FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_OFF to FPI_IMAGE_DEVICE_STATE_IDLE
(process:7507): libfprint-SSM-DEBUG: 15:47:33.876: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:33.894: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:33.899: 580.747760
(process:7507): libfprint-SSM-DEBUG: 15:47:33.899: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:33.905: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:33.911: 581.571135
(process:7507): libfprint-image-DEBUG: 15:47:33.927: Minutiae scan completed in 0.051966 secs
(process:7507): libfprint-device-DEBUG: 15:47:33.928: Device reported enroll progress, reported 5 of 5 have been completed
(process:7507): libfprint-DEBUG: 15:47:33.928: written to 'enrolled.pgm'
Wrote scanned image to enrolled.pgm
Enroll stage 5 of 5 passed. Yay!
(process:7507): libfprint-image_device-DEBUG: 15:47:33.928: Deactivating image device
(process:7507): libfprint-image_device-DEBUG: 15:47:33.928: Image device internal state change from FPI_IMAGE_DEVICE_STATE_IDLE to FPI_IMAGE_DEVICE_STATE_DEACTIVATING
(process:7507): libfprint-egis0575-DEBUG: 15:47:33.928: EGIS STOP
(process:7507): libfprint-SSM-DEBUG: 15:47:33.962: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:33.968: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:33.973: 580.549664
(process:7507): libfprint-SSM-DEBUG: 15:47:34.024: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:34.033: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:34.038: 582.226662
(process:7507): libfprint-SSM-DEBUG: 15:47:34.088: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:34.096: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:34.101: 579.538088
(process:7507): libfprint-SSM-DEBUG: 15:47:34.152: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:34.158: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:34.164: 581.914862
(process:7507): libfprint-SSM-DEBUG: 15:47:34.214: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:34.222: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:34.227: 579.422330
(process:7507): libfprint-SSM-DEBUG: 15:47:34.278: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:34.285: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:34.290: 579.013816
(process:7507): libfprint-SSM-DEBUG: 15:47:34.341: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:34.347: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:34.352: 581.729089
(process:7507): libfprint-SSM-DEBUG: 15:47:34.403: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:34.411: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-egis0575-DEBUG: 15:47:34.416: 578.467886
(process:7507): libfprint-SSM-DEBUG: 15:47:34.416: [egis0575] PACKET_SSM_DONE entering state 0
(process:7507): libfprint-SSM-DEBUG: 15:47:34.423: [egis0575] PACKET_SSM_DONE completed successfully
(process:7507): libfprint-device-DEBUG: 15:47:34.423: Device reported finger status change: FP_FINGER_STATUS_PRESENT
(process:7507): libfprint-image_device-DEBUG: 15:47:34.423: Image device reported finger status: on
(process:7507): libfprint-SSM-DEBUG: 15:47:34.423: [egis0575] IMG_SSM_DONE completed successfully
(process:7507): libfprint-image_device-DEBUG: 15:47:34.423: Image device deactivation completed
(process:7507): libfprint-image_device-DEBUG: 15:47:34.423: Image device internal state change from FPI_IMAGE_DEVICE_STATE_DEACTIVATING to FPI_IMAGE_DEVICE_STATE_INACTIVE
(process:7507): libfprint-device-DEBUG: 15:47:34.423: Device reported enroll completion
(process:7507): libfprint-device-DEBUG: 15:47:34.423: Device reported finger status change: FP_FINGER_STATUS_NONE
(process:7507): libfprint-device-DEBUG: 15:47:34.423: Print for finger FP_FINGER_LEFT_THUMB enrolled
(process:7507): libfprint-device-DEBUG: 15:47:34.423: Completing action FPI_DEVICE_ACTION_ENROLL in idle!
(process:7507): libfprint-device-DEBUG: 15:47:34.423: Updated temperature model after 4.70 seconds, ratio 0.27 -> 0.29, active 0 -> 0, FP_TEMPERATURE_WARM -> FP_TEMPERATURE_WARM
(process:7507): libfprint-DEBUG: 15:47:34.423: Device has not storage, saving print only locally
(process:7507): libfprint-egis0575-DEBUG: 15:47:34.446: EGIS DEINIT
(process:7507): libfprint-image_device-DEBUG: 15:47:34.446: Image device close completed
(process:7507): libfprint-device-DEBUG: 15:47:34.446: Device reported close completion
(process:7507): libfprint-device-DEBUG: 15:47:34.446: Completing action FPI_DEVICE_ACTION_CLOSE in idle!
(process:7507): libfprint-device-DEBUG: 15:47:34.446: Updated temperature model after 0.02 seconds, ratio 0.29 -> 0.29, active 0 -> 0, FP_TEMPERATURE_WARM -> FP_TEMPERATURE_WARM
```

**Comment by topni1** [comment]
Okey, the log looks normal...
I had also problems with the sensor "recognizing" a finger without it being there when the sensor was a little wet or anything was on it...
The best thing to look for the problem would be a usb package capture with wireshark...
but maybe a first step would be to see at the "enrolled.pgm" and see if it is just noise.

**Comment by ajboni** [comment]
![image](https://user-images.githubusercontent.com/12578313/161863266-292f1148-9ce5-4f0f-95ac-45d16e5fd39b.png)

Yes, looks like noise... BUT if  I start it with the finger on it it scans *something*  that looks like a fingerprint... (but only when I swipe it).

**Comment by topni1** [comment]
I have updated the repo with some changed values that i guess *could* change something so the sensor doesnt go off imidiatly...

**Comment by ajboni** [comment]
Better, I can finish the registration and verify loop but cant get a match.
Images (enrolled vs verify) look similar (but maybe flipped?")

**Comment by topni1** [comment]
Ok great that a finger comes from the driver that's good :)
I dont think they can be flipped (the code is the same)...
For the best enroll and verify i swipe the whole top part of the finger... maybe try out the ./image-capture to see at what speed it gets a good image

**Comment by pacifi5t** [comment]
Hi, the driver actually works on my laptop (Acer Swift 1 SF114-33 running Fedora 36 with GNOME). Here are pics for enrolled and verified respectively:
![enrolled](https://user-images.githubusercontent.com/57317100/177192330-97ebce44-b504-4a1e-917e-ec9ed2f8a280.png)
![verify](https://user-images.githubusercontent.com/57317100/177192346-49a2c9ff-645a-48f6-92cf-25e1d440da8e.png)

It even managed to verify finger:
```Print loaded. Time to verify!
(process:4667): libfprint-device-DEBUG: 19:04:47.295: Updated temperature model after 5.08 seconds, ratio 0.33 -> 0.33, active 1 -> 1, FP_TEMPERATURE_WARM -> FP_TEMPERATURE_WARM
(process:4667): libfprint-image_device-DEBUG: 19:04:47.295: Activating image device
(process:4667): libfprint-image_device-DEBUG: 19:04:47.295: Image device internal state change from FPI_IMAGE_DEVICE_STATE_INACTIVE to FPI_IMAGE_DEVICE_STATE_ACTIVATING
(process:4667): libfprint-egis0575-DEBUG: 19:04:47.295: EGIS START
(process:4667): libfprint-SSM-DEBUG: 19:04:47.295: [egis0575] image capture entering state 0
(process:4667): libfprint-egis0575-DEBUG: 19:04:47.295: EGIS START DONE
(process:4667): libfprint-image_device-DEBUG: 19:04:47.295: Image device activation completed
(process:4667): libfprint-image_device-DEBUG: 19:04:47.296: Image device internal state change from FPI_IMAGE_DEVICE_STATE_ACTIVATING to FPI_IMAGE_DEVICE_STATE_IDLE
(process:4667): libfprint-image_device-DEBUG: 19:04:47.296: Image device internal state change from FPI_IMAGE_DEVICE_STATE_IDLE to FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_ON
(process:4667): libfprint-device-DEBUG: 19:04:47.296: Device reported finger status change: FP_FINGER_STATUS_NEEDED
(process:4667): libfprint-SSM-DEBUG: 19:04:47.297: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:47.320: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-egis0575-DEBUG: 19:04:47.326: 20.236744
(process:4667): libfprint-device-DEBUG: 19:04:47.326: Device reported finger status change: FP_FINGER_STATUS_NEEDED | FP_FINGER_STATUS_PRESENT
(process:4667): libfprint-image_device-DEBUG: 19:04:47.326: Image device reported finger status: on
(process:4667): libfprint-image_device-DEBUG: 19:04:47.326: Image device internal state change from FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_ON to FPI_IMAGE_DEVICE_STATE_CAPTURE
(process:4667): libfprint-SSM-DEBUG: 19:04:47.326: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:47.333: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-egis0575-DEBUG: 19:04:47.339: 20.063480
(process:4667): libfprint-image_device-DEBUG: 19:04:47.339: Image device reported finger status: on
(process:4667): libfprint-SSM-DEBUG: 19:04:47.339: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:47.345: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-egis0575-DEBUG: 19:04:47.351: 19.709111
(process:4667): libfprint-image_device-DEBUG: 19:04:47.351: Image device reported finger status: on
(process:4667): libfprint-SSM-DEBUG: 19:04:47.351: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:47.358: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-egis0575-DEBUG: 19:04:47.364: 20.235997
(process:4667): libfprint-image_device-DEBUG: 19:04:47.364: Image device reported finger status: on
(process:4667): libfprint-SSM-DEBUG: 19:04:47.364: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:47.371: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-egis0575-DEBUG: 19:04:47.376: 20.040702
(process:4667): libfprint-image_device-DEBUG: 19:04:47.376: Image device reported finger status: on
(process:4667): libfprint-SSM-DEBUG: 19:04:47.376: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:47.383: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-egis0575-DEBUG: 19:04:47.388: 20.294810
(process:4667): libfprint-egis0575-DEBUG: 19:04:47.388: To many images without Finger.
(process:4667): libfprint-device-DEBUG: 19:04:47.388: Device reported finger status change: FP_FINGER_STATUS_NEEDED
(process:4667): libfprint-image_device-DEBUG: 19:04:47.389: Image device reported finger status: off
(process:4667): libfprint-SSM-DEBUG: 19:04:47.389: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:47.396: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-SSM-DEBUG: 19:04:49.286: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:49.310: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-egis0575-DEBUG: 19:04:49.316: 180.546677
(process:4667): libfprint-SSM-DEBUG: 19:04:49.316: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:49.323: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-egis0575-DEBUG: 19:04:49.328: 198.986370
(process:4667): libfprint-SSM-DEBUG: 19:04:49.378: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:49.385: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-egis0575-DEBUG: 19:04:49.390: 226.060866
(process:4667): libfprint-SSM-DEBUG: 19:04:49.441: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:49.447: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-egis0575-DEBUG: 19:04:49.452: 233.720874
(process:4667): libfprint-SSM-DEBUG: 19:04:49.502: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:49.509: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-egis0575-DEBUG: 19:04:49.515: 235.664488
(process:4667): libfprint-SSM-DEBUG: 19:04:49.565: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:49.571: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-egis0575-DEBUG: 19:04:49.577: 238.082151
(process:4667): libfprint-SSM-DEBUG: 19:04:49.627: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:49.636: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-egis0575-DEBUG: 19:04:49.642: 239.800784
(process:4667): libfprint-SSM-DEBUG: 19:04:49.692: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:49.698: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-egis0575-DEBUG: 19:04:49.704: 241.504108
(process:4667): libfprint-SSM-DEBUG: 19:04:49.754: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:49.760: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-egis0575-DEBUG: 19:04:49.766: 243.425878
(process:4667): libfprint-SSM-DEBUG: 19:04:49.816: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:49.823: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-egis0575-DEBUG: 19:04:49.829: 243.677558
(process:4667): libfprint-SSM-DEBUG: 19:04:49.829: [egis0575] PACKET_SSM_DONE entering state 0
(process:4667): libfprint-SSM-DEBUG: 19:04:49.836: [egis0575] PACKET_SSM_DONE completed successfully
(process:4667): libfprint-assembling-DEBUG: 19:04:49.994: calc delta completed in 0.158229 secs
(process:4667): libfprint-assembling-DEBUG: 19:04:50.122: calc delta completed in 0.127434 secs
(process:4667): libfprint-assembling-DEBUG: 19:04:50.122: errors: 104254 rev: 110916
(process:4667): libfprint-assembling-DEBUG: 19:04:50.250: calc delta completed in 0.127497 secs
(process:4667): libfprint-assembling-DEBUG: 19:04:50.250: height is 387
(process:4667): libfprint-image_device-DEBUG: 19:04:50.250: Image device captured an image
(process:4667): libfprint-image_device-DEBUG: 19:04:50.250: Image device internal state change from FPI_IMAGE_DEVICE_STATE_CAPTURE to FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_OFF
(process:4667): libfprint-device-DEBUG: 19:04:50.250: Device reported finger status change: FP_FINGER_STATUS_NONE
(process:4667): libfprint-image_device-DEBUG: 19:04:50.250: Image device reported finger status: off
(process:4667): libfprint-image_device-DEBUG: 19:04:50.250: Image device internal state change from FPI_IMAGE_DEVICE_STATE_AWAIT_FINGER_OFF to FPI_IMAGE_DEVICE_STATE_IDLE
(process:4667): libfprint-image_device-DEBUG: 19:04:50.250: Deactivating image device
(process:4667): libfprint-image_device-DEBUG: 19:04:50.250: Image device internal state change from FPI_IMAGE_DEVICE_STATE_IDLE to FPI_IMAGE_DEVICE_STATE_DEACTIVATING
(process:4667): libfprint-egis0575-DEBUG: 19:04:50.250: EGIS STOP
(process:4667): libfprint-SSM-DEBUG: 19:04:50.250: [egis0575] image capture entering state 10
(process:4667): libfprint-image_device-DEBUG: 19:04:50.250: Image device deactivation completed
(process:4667): libfprint-image_device-DEBUG: 19:04:50.250: Image device internal state change from FPI_IMAGE_DEVICE_STATE_DEACTIVATING to FPI_IMAGE_DEVICE_STATE_INACTIVE
(process:4667): libfprint-image-DEBUG: 19:04:50.280: Minutiae scan completed in 0.029609 secs
(process:4667): libfprint-print-DEBUG: 19:04:50.280: score 0/15
(process:4667): libfprint-print-DEBUG: 19:04:50.281: score 0/15
(process:4667): libfprint-print-DEBUG: 19:04:50.281: score 0/15
(process:4667): libfprint-print-DEBUG: 19:04:50.281: score 0/15
(process:4667): libfprint-print-DEBUG: 19:04:50.281: score 0/15
(process:4667): libfprint-print-DEBUG: 19:04:50.282: score 24/15
(process:4667): libfprint-device-DEBUG: 19:04:50.282: Device reported verify result
(process:4667): libfprint-DEBUG: 19:04:50.283: written to 'verify.pgm'
Print image saved as verify.pgm
(process:4667): libfprint-DEBUG: 19:04:50.283: Match report: device LighTuning Technology Inc. EgisTec EH575 matched finger right index successifully with print (null), enrolled on date 2022-07-04 by user root
MATCH!
(process:4667): libfprint-device-DEBUG: 19:04:50.283: Device reported verify completion
(process:4667): libfprint-device-DEBUG: 19:04:50.283: Completing action FPI_DEVICE_ACTION_VERIFY in idle!
(process:4667): libfprint-device-DEBUG: 19:04:50.283: Updated temperature model after 2.99 seconds, ratio 0.33 -> 0.34, active 0 -> 0, FP_TEMPERATURE_WARM -> FP_TEMPERATURE_WARM
```
Though, I don't know if I swipe correctly (from left to right), so verification is false negative sometimes.
The scanner: `Bus 001 Device 003: ID 1c7a:0575 LighTuning Technology Inc. EgisTec EH575`

**Comment by f-klement** [comment]
@pacifi5t could you please share your process on how you got it to work?

**Comment by topni1** [comment]
@[pacifi5t](https://github.com/pacifi5t) Swipe direction is from downwards (so the movement is from the keyboard down to the edge).

**Comment by topni1** [comment]
@klementf95
You need to clone [this](https://gitlab.freedesktop.org/topni1/libfprint.git) repo.
Then you navigate into the main folder and execute the commands like shown in the second comment in this issue.

If you or anyone has any issues with the driver, please make a new issue [here](https://gitlab.freedesktop.org/topni1/libfprint/-/issues), because that is the driver that is currently being merged into libfprint.
I am not sure @Animeshz still works on this github repo, because I think he does not have access to a egis0575 anymore.

**Comment by pacifi5t** [comment]
> @[pacifi5t](https://github.com/pacifi5t) Swipe direction is from downwards (so the movement is from the keyboard down to the edge).

Ok, thank you. I'll try it this way than.

**Comment by pacifi5t** [comment]
@klementf95
I did nothing special. There's an instruction in the second message in this issue. The only problem is that the list of dependencies is incomplete, but meson will tell you what are missing (if you're using Fedora, don't forget to install packages with "-devel" suffix).
And the device is still busy after exiting the program, so you'll have to wait some time.

**Comment by f-klement** [comment]
I got through most of the steps, but failed to verify:

sudo ./verify
(process:111924): libfprint-context-DEBUG: 22:28:08.424: Initializing FpContext (libfprint version 1.94.4)
libusb: warning [libusb_init] installing new context as implicit default
(process:111924): libfprint-context-DEBUG: 22:28:08.428: No driver found for USB device 1D6B:0003
(process:111924): libfprint-context-DEBUG: 22:28:08.428: No driver found for USB device 1D6B:0002
(process:111924): libfprint-context-DEBUG: 22:28:08.428: No driver found for USB device 1D6B:0003
(process:111924): libfprint-context-DEBUG: 22:28:08.428: No driver found for USB device 04CA:3802
(process:111924): libfprint-context-DEBUG: 22:28:08.428: No driver found for USB device 04F2:B6DD
(process:111924): libfprint-context-DEBUG: 22:28:08.428: No driver found for USB device 1D6B:0002
(process:111924): libfprint-device-DEBUG: 22:28:08.428: Device reported probe completion
(process:111924): libfprint-device-DEBUG: 22:28:08.428: Completing action FPI_DEVICE_ACTION_PROBE in idle!
(process:111924): libfprint-device-DEBUG: 22:28:08.428: Updated temperature model after 0.00 seconds, ratio 0.27 -> 0.27, active 0 -> 0, FP_TEMPERATURE_COLD -> FP_TEMPERATURE_COLD
Selected device 0 (LighTuning Technology Inc. EgisTec EH575) claimed by egis0575 driver
(process:111924): libfprint-egis0575-DEBUG: 22:28:08.674: EGIS INIT
(process:111924): libfprint-SSM-DEBUG: 22:28:08.674: [egis0575] SSM_SETUP_DONE entering state 0
(process:111924): libfprint-SSM-DEBUG: 22:28:08.674: [egis0575] SSM_SETUP_DONE entering state 1
(process:111924): libfprint-SSM-DEBUG: 22:28:08.674: [egis0575] PACKET_SSM_DONE entering state 0
(process:111924): libfprint-SSM-DEBUG: 22:28:08.699: [egis0575] PACKET_SSM_DONE completed successfully
(process:111924): libfprint-SSM-DEBUG: 22:28:08.699: [egis0575] SSM_SETUP_DONE entering state 2
(process:111924): libfprint-SSM-DEBUG: 22:28:08.699: [egis0575] SSM_SETUP_DONE entering state 3
(process:111924): libfprint-SSM-DEBUG: 22:28:08.704: [egis0575] SSM_SETUP_DONE completed successfully
(process:111924): libfprint-egis0575-DEBUG: 22:28:08.704: Setup calibration package is broken, retry in some time.
(process:111924): libfprint-image_device-DEBUG: 22:28:08.704: Image device open completed
(process:111924): libfprint-device-DEBUG: 22:28:08.704: Device reported open completion
(process:111924): libfprint-device-DEBUG: 22:28:08.705: Completing action FPI_DEVICE_ACTION_OPEN in idle!
(process:111924): libfprint-device-DEBUG: 22:28:08.705: Updated temperature model after 0.28 seconds, ratio 0.27 -> 0.27, active 0 -> 0, FP_TEMPERATURE_COLD -> FP_TEMPERATURE_COLD

(process:111924): libfprint-WARNING **: 22:28:08.705: Failed to open device: Setup calibration package broken, retry later
libusb: warning [libusb_exit] device 3.3 still referenced
libusb: warning [libusb_exit] device 3.1 still referenced
libusb: warning [libusb_exit] application left some devices open

**Comment by pacifi5t** [comment]
@klementf95
I mentioned this problem in my last comment:
> And the device is still busy after exiting the program, so you'll have to wait some time.

Unfortunately, I don't know the exact time when it will be available again.

**Comment by topni1** [comment]
Yeah, you can just try over and over again, it should work sometimes at least...
I am waiting on a feature of libfprint to get this fixed. If that is implemented the driver should work every time. Hope it will be not that long until that happens.

**Comment by f-klement** [comment]
Any advice from your experience on how to make the verification results more consistent? Apart from swiping downward that is.

**Comment by topni1** [comment]
slower swipe and experimentation while looking at the verified.pgm file, or just to get a feel for it, use imgcapture and look into the finger.pgm file how that looks

**Comment by pacifi5t** [comment]
@topni1
I opened an issue at your GitLab:
> If you or anyone has any issues with the driver, please make a new issue [here](https://gitlab.freedesktop.org/topni1/libfprint/-/issues), because that is the driver that is currently being merged into libfprint.

Could you please view it?

**Comment by mute-09** [comment]
I have installed libfprint-egis-0575 with yay on an Acer Swift 3x laptop (SF314-510G) with device 1c7a:0575 and my fingerprint scanner can detect, I use arch linux with kde plasma 6 :

> fprintd-verify
> Using device /net/reactivated/Fprint/Device/0 Listing enrolled fingers:
> #0: right-index-finger
> #1: right-thumb Verify started! Verifying: right-index-finger
> Verify result: verify-match (done)
